logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageMapper.kt

```diff
@@ -96,6 +94,32 @@ class MessageMapperImpl(
                 visibility = visibility,
                 senderName = message.senderUserName,
             )
+
+            is Message.Ephemeral -> {
+                with(message.ephemeralMessage) {
+                    MessageEntity.Ephemeral(
+                        expireAfterMillis = message.expireAfterMillis,
+                        selfDeletionStartDate = message.selfDeletionStartDate,
+                        ephemeralMessage = MessageEntity.Regular(
+                            id = id,
+                            content = toMessageEntityContent(content),
+                            conversationId = conversationId.toDao(),
+                            date = date.toInstant(),
+                            senderUserId = senderUserId.toDao(),
+                            senderClientId = senderClientId.value,
+                            status = status,
+                            editStatus = when (editStatus) {
+                                is Message.EditStatus.NotEdited -> MessageEntity.EditStatus.NotEdited
+                                is Message.EditStatus.Edited -> MessageEntity.EditStatus.Edited(editStatus.lastTimeStamp.toInstant())
+                            },
+                            visibility = visibility,
+                            senderName = senderUserName,
+                            isSelfMessage = isSelfMessage,
+                            expectsReadConfirmation = expectsReadConfirmation
+                        )
+                    )
+                }
+            }
```

Comment:
This whole mapping could be removed and we could just the `ephemeralData` as part of the `regular` message.